src/main/java/uk/gov/hmcts/dts/fact/services/CourtService.java

```diff
@@ -126,6 +136,20 @@ public Court getCourtBySlug(final String slug) {
             .collect(toList());
     }
 
+    public List<CourtReferenceWithDistance> getNearestCourtReferencesByPostcode(final String postcode) {
+        final Optional<MapitData> optionalMapitData = mapitService.getMapitData(postcode);
+
+        if (optionalMapitData.isEmpty()) {
+            LOGGER.error("No mapit data found for provided postcode: {}", postcode);
+            return new ArrayList<>(); // Return empty so the frontend logic can be invoked
+        }
+
+        List<CourtReferenceWithDistance> crwds = convert(proximitySearch.searchWith(optionalMapitData.get()));
```

Comment:
Sorry to be fussy. Can we use full variable names rather than short form like this one?